config/index.ts

```diff
@@ -1,268 +1,406 @@
-import { CmsPublishMode } from '../types/Files';
+import fs from 'fs-extra';
 
-// Use new config if it exists
-export function loadConfig(
-  path: string,
-  options: CLIOptions = {}
-): CLIConfig | null {
-  // Attempt to load the root config
-  if (newConfigFileExists()) {
-    return CLIConfiguration.init(options);
-  }
-  return config_DEPRECATED.loadConfig(path, options);
+import { MIN_HTTP_TIMEOUT } from '../constants/config';
+import { HubSpotConfigAccount } from '../types/Accounts';
+import { HubSpotConfig, ConfigFlag } from '../types/Config';
+import { CmsPublishMode } from '../types/Files';
+import { logger } from '../lib/logger';
+import {
+  getGlobalConfigFilePath,
+  getLocalConfigFilePath,
+  readConfigFile,
+  parseConfig,
+  buildConfigFromEnvironment,
+  writeConfigFile,
+  getLocalConfigFileDefaultPath,
+  getConfigAccountByIdentifier,
+  isConfigAccountValid,
+  getConfigAccountIndexById,
+  getConfigPathEnvironmentVariables,
+  getConfigAccountByInferredIdentifier,
+} from './utils';
+import { CMS_PUBLISH_MODE } from '../constants/files';
+import { Environment } from '../types/Config';
+import { i18n } from '../utils/lang';
+export function localConfigFileExists(): boolean {
+  return Boolean(getLocalConfigFilePath());
 }
 
-export function getAndLoadConfigIfNeeded(
-  options?: CLIOptions
-): Partial<CLIConfig> | null {
-  if (CLIConfiguration.isActive()) {
-    return CLIConfiguration.config;
-  }
-  return config_DEPRECATED.getAndLoadConfigIfNeeded(options);
+export function globalConfigFileExists(): boolean {
+  return fs.existsSync(getGlobalConfigFilePath());
 }
 
-export function validateConfig(): boolean {
-  if (CLIConfiguration.isActive()) {
-    return CLIConfiguration.validate();
-  }
-  return config_DEPRECATED.validateConfig();
-}
+function getDefaultConfigFilePath(): string {
+  const globalConfigFilePath = getGlobalConfigFilePath();
 
-export function loadConfigFromEnvironment(): boolean {
-  if (CLIConfiguration.isActive()) {
-    return CLIConfiguration.useEnvConfig;
+  if (fs.existsSync(globalConfigFilePath)) {
+    return globalConfigFilePath;
   }
-  return Boolean(config_DEPRECATED.loadConfigFromEnvironment());
-}
 
-export function createEmptyConfigFile(
-  options: { path?: string } = {},
-  useHiddenConfig = false
-): void {
-  if (useHiddenConfig) {
-    CLIConfiguration.write({ accounts: [] });
-  } else {
-    return config_DEPRECATED.createEmptyConfigFile(options);
+  const localConfigFilePath = getLocalConfigFilePath();
+
+  if (!localConfigFilePath) {
+    throw new Error(i18n('config.getDefaultConfigFilePath.error'));
   }
+
+  return localConfigFilePath;
 }
 
-export function deleteEmptyConfigFile() {
-  if (CLIConfiguration.isActive()) {
-    return CLIConfiguration.delete();
-  }
-  return config_DEPRECATED.deleteEmptyConfigFile();
+export function getConfigFilePath(): string {
+  const { configFilePathFromEnvironment } = getConfigPathEnvironmentVariables();
```

Comment:
`getConfigFilePath` checks for a `HUBSPOT_CONFIG_PATH` environment variable if the user is using a custom path - this variable will need to be set by the CLI if the user passes in the `--config` flag